src/www/widgets/widgets/gateways.widget.php

```diff
@@ -58,9 +58,10 @@
 <script>
   function gateways_widget_update(sender, data)
   {
+      var tbody = sender.find('tbody');
       data.map(function(gateway) {
           var tr_id = "gateways_widget_gw_" + gateway['name'];
-          if (find("#"+tr_id).length != 0) {
+          if (tbody.find("#"+tr_id).length != 0) {
```

Comment:
for me, this looks like a workaround for a bug (id is not unique), which would be a better idea to fix.

Reply:
maybe a browser-specific fix. we don't have tbody in the original file. maybe find without an object misses the point of searching for a html element

Reply:
@fichtner the tbody may be automatically generated by the browser (I have seen that multiple times in the developer tools). An id must be unique in a document by the HTML specification.

Reply:
yep, and this can fail when you use child selectors. IDs shouldn't be duplicated here, it just misses finding any ID so status stays at "~". sender.find("#"+tr_id) should work too?!